### PR TITLE
fix: 10116 intake clearing

### LINKED ISSUE #10116: Missing non-expired self parent in `FCM-MultiSBReconnect-3R-1k-20m`
Exception occurs in `Shadowgraph`:

```
Missing non-expired self parent for (0,6223,11c62181) sp(0,6142,7fe1967a) op(5,6222,8b13025b)
```

- Failure is consistent. [data folder](http://35.247.76.217:8095/swirlds-automation/develop/10N/MultiSBReconnect/20231127-055648-GCP-Daily-MultiSBReconnect-3R-10N/FCM-MultiSBReconnect-3R-1k-20m/)

Presumably related to https://github.com/hashgraph/hedera-services/pull/9747

### PR DESCRIPTION
**Pull request by alittley**
- closes #10116 
- [wiring diagram](https://mermaid.ink/svg/Zmxvd2NoYXJ0IExSCnYwKCgiKiIpKQpzdWJncmFwaCB2MVsiQ29uc2Vuc3VzIFBpcGVsaW5lIl0Kc3ViZ3JhcGggdjJbIkxpbmtlZCBFdmVudCBJbnRha2UiXQp2M3t7ImdldE1pbmltdW1HZW5lcmF0aW9uTm9uQW5jaWVudCJ9fQp2NFsibGlua2VkRXZlbnRJbnRha2UiXQp2NXt7ImxpbmtlZEV2ZW50SW50YWtlX3NwbGl0dGVyIn19CmVuZAp2NlsiaW5PcmRlckxpbmtlcjxiciAvPioiXQplbmQKc3ViZ3JhcGggdjdbIkludGFrZSBQaXBlbGluZSJdCnN1YmdyYXBoIHY4WyJFdmVudCBWYWxpZGF0aW9uIl0KdjlbImV2ZW50RGVkdXBsaWNhdG9yPGJyIC8+KiJdCnYxMFsiZXZlbnRTaWduYXR1cmVWYWxpZGF0b3I8YnIgLz4qIl0KdjExWyJpbnRlcm5hbEV2ZW50VmFsaWRhdG9yIl0KZW5kCnN1YmdyYXBoIHYxMlsiT3JwaGFuIEJ1ZmZlciJdCnYxM1sib3JwaGFuQnVmZmVyPGJyIC8+KiJdCnYxNHt7Im9ycGhhbkJ1ZmZlcl9zcGxpdHRlciJ9fQplbmQKZW5kCnYxNXt7ImV4dHJhY3Qgb2xkZXN0TWluaW11bUdlbmVyYXRpb25PbkRpc2sifX0KdjE2WyJzaWduZWRTdGF0ZUZpbGVNYW5hZ2VyIl0KdjE3e3sidG8gU3RhdGVXcml0dGVuVG9EaXNrQWN0aW9uIn19CnYxOCgoIuKZoSIpKQp2OSAtLSAiZXZlbnRzIHdpdGggdW52YWxpZGF0ZWQgc2lnbmF0dXJlcyIgLS0+IHYxMAp2MTAgLS0gInVub3JkZXJlZCBldmVudHMiIC0tPiB2MTMKdjMgLS4gIm1pbmltdW0gZ2VuZXJhdGlvbiBub24gYW5jaWVudCIgLi0+IHYwCnY2IC0tICJsaW5rZWQgZXZlbnRzIiAtLT4gdjQKdjExIC0tICJub24tZGVkdXBsaWNhdGVkIGV2ZW50cyIgLS0+IHY5CnY0IC0tPiB2NQp2NSAtLT4gdjMKdjEzIC0tPiB2MTQKdjE0IC0tICJ1bmxpbmtlZCBldmVudHMiIC0tPiB2Ngp2MTYgLS0+IHYxNQp2MTYgLS0+IHYxNwpjbGFzc0RlZiBzMCBmaWxsOiNmODgsc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MCx2MTggczAKY2xhc3NEZWYgczEgZmlsbDojOUNGLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjEsdjcgczEKY2xhc3NEZWYgczIgZmlsbDojQkVGLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjEyLHYyLHY4IHMyCmNsYXNzRGVmIHMzIGZpbGw6I2NjYyxzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxNCx2MTUsdjE3LHYzLHY1IHMzCmNsYXNzRGVmIHM0IGZpbGw6I2ZmOSxzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxMCx2MTEsdjEzLHYxNix2NCx2Nix2OSBzNAo=?bgColor=e8e8e8)

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/deduplication/EventDeduplicator.java [hunk 1]
```diff
@@ -145,4 +146,13 @@ public void setMinimumGenerationNonAncient(final long minimumGenerationNonAncien
 
         observedEvents.shiftWindow(minimumGenerationNonAncient);
     }
+
+    /**
+     * Clear the internal state of this deduplicator.
+     *
+     * @param ignored ignored trigger object
+     */
+    public void clear(final ClearTrigger ignored) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/deduplication/EventDeduplicator.java:155]:
```suggestion
    public void clear(@NonNull final ClearTrigger ignored) {
```

**Review by cody-littley — DISMISSED** [review]

**Review by cody-littley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/ClearTrigger.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.wiring;
+
+/**
+ * A placeholder object, to provide a type to the input `clear` wire for wiring components.
+ */
+public record ClearTrigger() {}
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/ClearTrigger.java:22]: Are we planning to have multiple control commands like this where we create records, whether empty or with parameters? If we have multiple commands common across multiple components, we might want a generic InputWire where the Command Interface defines all the command records that are usable with it.

I'm just thinking about how we would generalize this pattern.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformCoordinator.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        inOrderLinkerWiring.flushRunnable().run();
+        linkedEventIntakeWiring.flushRunnable().run();
+
+        // once everything has been flushed out of the system, it's safe to unpause the linked event intake
+        linkedEventIntakeWiring.pauseInput().inject(false);
+
+        // data is no longer moving through the system. clear all the internal data structures in the wiring objects
+        eventDeduplicatorWiring.clearInput().inject(new ClearTrigger());
+        eventDeduplicatorWiring.flushRunnable().run();
+        orphanBufferWiring.clearInput().inject(new ClearTrigger());
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformCoordinator.java:83]: Is there a reason the ClearTrigger is created multiple times as opposed to being a static singleton?

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/ClearTrigger.java (hunk 2 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/ClearTrigger.java:22]: There will definitely be other instances of this pattern. I've observed at least one other case while working on PCES, also containing no parameters.

I can't think of any benefits off the top of my head from making a common command interface, but I'm not opposed to the idea if there's a clean way that makes things easier to understand.

**Review by edward-swirldslabs — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Pull request merged**